enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,554 @@
+---
+title: hcp-etcd-backup-api
+
+authors:
+  - "@jparrill"
+
+reviewers:
+  - "@csrwng"
+  - "@enxebre"
+  - "@sjenning"
+  - "@derekwaynecarr"
+  - "@imain"
+  - "@bryan-cox"
+  - "@jparrill"
+  - "@muraee"
+  - "@Patryk-Stefanski"
+  - "@celebdor"
+
+approvers:
+  - "@csrwng"
+  - "@enxebre"
+  - "@sjenning"
+  - "@derekwaynecarr"
+  - "@imain"
+  - "@bryan-cox"
+  - "@jparrill"
+  - "@muraee"
+  - "@Patryk-Stefanski"
+  - "@celebdor"
+
+api-approvers:
+  - "@csrwng"
+  - "@enxebre"
+  - "@sjenning"
+  - "@derekwaynecarr"
+  - "@imain"
+  - "@bryan-cox"
+  - "@jparrill"
+  - "@muraee"
+  - "@Patryk-Stefanski"
+  - "@celebdor"
+
+
+creation-date: 2024-02-01
+last-updated: 2024-02-16
+
+tracking-link:
+  - https://issues.redhat.com/browse/HOSTEDCP-1370
+
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Hosted Control Plane Etcd Backup API
+
+## Summary
+
+This proposal fleshes out the details for the new automated Etcd Backup API solution for Hosted Control Planes.
+
+## Glossary
+
+- HCP - Hosted Control Plane
+- HC - Hosted Cluster
+- PVC - Persistent Volume Claim
+- PV - Persistent Volume
+
+## Motivation
+
+Hypershift now has a small interface to perform the etcd backups using the CLI. This solution is not fully integrated into the product and developing an API will help to grow the product in many senses.
+
+The backup API provides a standardized and documented interface, making it easier for developers and customers to understand and implement backup functionalities for their Hosted Clusters. This not only accelerates the integration process but also encourages the development of a robust ecosystem around your product.
+
+Contributes significantly to the reliability of the product by promoting seamless integration, automation, and standardization, ultimately providing users with a more dependable and user-friendly experience.
+
+The API will help with disaster recovery duties, like Hosted Cluster migration or Hosted Cluster recovery, both highly demanded features by Red Hat customers.
+
+### User Stories
+
+- As a customer I want to perform etcd backups of my Hosted Clusters in a concrete time window and schedule at the pace my business needs.
+- As a customer/engineer/service provider I want to have the ability to restore etcd cluster in a new hosted cluster.
+- As a developer I want to declaratively create etcd backups using and upload it to Amazon S3.
+- As a developer I want to declaratively create etcd backups using and upload it to Azure.
+- As a customer I want to perform etcd backups of my Hosted Clusters in a concrete time window and schedule in the pace my bussiness needs.
```

Comment:
What about agent, disconnected, and kubevirt HCs?

Reply:
Added a new use case. Thanks